src/bsoncxx/types/bson_value/value.cpp

```diff
@@ -77,12 +77,28 @@ value::value(bool v) : _impl{stdx::make_unique<impl>()} {
     _impl->_value.value.v_bool = v;
 }
 
-value::value(b_code v) : value(v.type_id, v) {}
-value::value(b_regex v) : value(v.type_id, std::string{v.regex}, std::string{v.options}) {}
-value::value(b_symbol v) : value(v.type_id, v) {}
 value::value(b_maxkey) : value(type::k_maxkey) {}
 value::value(b_minkey) : value(type::k_minkey) {}
 value::value(b_undefined) : value(type::k_undefined) {}
+value::value(const type id) : _impl{stdx::make_unique<impl>()} {
+    switch (id) {
+        case type::k_minkey:
+            _impl->_value.value_type = BSON_TYPE_MINKEY;
+            break;
+        case type::k_maxkey:
+            _impl->_value.value_type = BSON_TYPE_MAXKEY;
+            break;
+        case type::k_undefined:
+            _impl->_value.value_type = BSON_TYPE_UNDEFINED;
+            break;
+        default:
+            throw bsoncxx::exception(error_code::k_invalid_bson_type_id);
+    }
+}
+
+value::value(b_code v) : value(v.type_id, v) {}
+value::value(b_symbol v) : value(v.type_id, v) {}
```

Comment:
I suggest splitting off code/symbol from regex for the same reason as splitting of the minkey/maxkey/undefined constructors (so `bson_value::value{type::k_symbol, "x", "y"}` does not compile and run and ignore the second argument).

Reply:
Done